packages/react-dom/src/shared/DOMProperty.js

```diff
@@ -201,6 +176,54 @@ export function isReservedProp(name) {
   return RESERVED_PROPS.hasOwnProperty(name);
 }
 
+const attributeNames = new Map([
+  ['acceptCharset', 'accept-charset'],
+  ['className', 'class'],
+  ['htmlFor', 'for'],
+  ['httpEquiv', 'http-equiv'],
+
+  ['autoReverse', 'autoReverse'],
+  ['externalResourcesRequired', 'externalResourcesRequired'],
+  ['preserveAlpha', 'preserveAlpha'],
+]);
+
+export function getAttributeName(name) {
+  if (attributeNames.has(name)) {
+    return attributeNames.get(name);
+  }
+  // TODO: it is odd that this depends on `properties` whitelist.
+  const attributeName = properties.hasOwnProperty(name)
+    ? name.toLowerCase()
+    : name;
+  attributeNames.set(name, attributeName);
+  return attributeName;
+}
+
+export function getAttributeNamespace(name) {
+  switch (name) {
+    case 'xlinkActuate':
+    case 'xlinkArcrole':
+    case 'xlinkHref':
+    case 'xlinkRole':
+    case 'xlinkShow':
+    case 'xlinkTitle':
+    case 'xlinkType':
+      return 'http://www.w3.org/1999/xlink';
+    case 'xmlBase':
+    case 'xmlLang':
+    case 'xmlSpace':
+      return 'http://www.w3.org/XML/1998/namespace';
+    default:
+      return null;
+  }
+}
+
+const usePropertiesFor = new Set(['checked', 'multiple', 'muted', 'selected']);
+
+export function shouldUseProperty(name) {
+  return usePropertiesFor.has(name);
+}
```

Comment:
Could this just be inlined in `DOMPropertyOperations`? Doesn't look like it's used anywhere else. Also, we could then just do check the set directly instead of a function call too.

Reply:
I want to consolidate all property/attribute stuff here first, then refactor it to make function contracts clearer, then possibly spread it out. I'd rather not spread it out yet.

Reply:
It's just easier for me to deal with it if it's all in one file. Easier to see patterns.